src/templater.rs

```diff
@@ -561,6 +561,38 @@ mod tests {
     }
 }
 
+pub struct IdWithHighlightedPrefix {
+    prefix: String,
+    rest: String,
+}
+
+impl Template<()> for IdWithHighlightedPrefix {
+    fn format(&self, _: &(), formatter: &mut dyn Formatter) -> io::Result<()> {
+        formatter.with_label("prefix", |fmt| fmt.write_str(&self.prefix))?;
+        formatter.with_label("rest", |fmt| fmt.write_str(&self.rest))
+    }
+}
+
+pub struct HighlightPrefix;
+impl TemplateProperty<CommitOrChangeId<'_>> for HighlightPrefix {
+    type Output = IdWithHighlightedPrefix;
+
+    fn extract(&self, context: &CommitOrChangeId) -> Self::Output {
```

Comment:
(No action needed)

I'll move this to `CommitOrChangeId::shortest_styled_prefix()` after #1149.

Reply:
Ah, that sounds great, thanks!